### PR TITLE
feat: deny permissions

### PR DESCRIPTION
**Pull request by pavanjoshi914**
### Describe the changes you have made in this PR

deny permissoins functionality

### Type of change

(Remove other not matching type)

- `feat`: New feature (non-breaking change which adds functionality)

### Checklist

- [X] Self-review of changed code
- [X] Manual testing
- [X] Added automated tests where applicable
- [X] Update Docs & Guides
- For UI-related changes
- [X] Darkmode
- [X] Responsive layout

### CODE REVIEW

File: src/app/screens/Nostr/ConfirmSignMessage.tsx [hunk 1]
```diff
@@ -137,10 +147,10 @@ function ConfirmSignMessage() {
                   }),
                 })}
               />
-              <ConfirmOrCancel
+              <SignOrDeny
                 disabled={loading}
                 loading={loading}
-                onCancel={reject}
+                onDeny={reject}
```
Comment (reneaaron) [inline src/app/screens/Nostr/ConfirmSignMessage.tsx:153]: I think there is a difference between closing the window and actively clicking deny. 

Closing the window = Cancel the request
Deny = Cancel the request + add a block permission

Does that make sense @stackingsaunter ?

**Review by reneaaron — CHANGES_REQUESTED** [review]
Great progress! :100: 

cACK, will need to do some more testing

+ We might want to add some unit tests for the new block feature?

File: src/app/components/SitePreferences/index.tsx [hunk 2]
```diff
@@ -318,6 +318,82 @@ function SitePreferences({ launcherType, allowance, onEdit, onDelete }: Props) {
...
+                </p>
+                <div>
+                  {permissions
+                    .filter((x) => x.enabled && x.blocked)
+                    .map((permission) => (
+                      <>
+                        <Badge
+                          key={permission.method}
+                          label={
+                            permission.method
```
[the remark this answers is not in the document]
Comment (pavanjoshi914) [reply src/app/components/SitePreferences/index.tsx:333]: i didn't understood this fully? do you mean i18n key? we already have function for that

[the remark this answers is not in the document]
Comment (reneaaron) [reply src/app/components/SitePreferences/index.tsx:333]: Now we have 2 huge code blocks that do nothing else than finding the permission translations. Let's refactor that into a method and use this method in both places...

[the remark this answers is not in the document]
Comment (pavanjoshi914) [reply src/app/components/SitePreferences/index.tsx:333]: resolved

File: src/app/screens/Nostr/ConfirmSignMessage.tsx (hunk 1 above)
Comment (stackingsaunter) [reply src/app/screens/Nostr/ConfirmSignMessage.tsx:153]: make sense

**Review by reneaaron — CHANGES_REQUESTED** [review]

File: src/extension/background-script/actions/nostr/enable.ts [hunk 3]
```diff
@@ -108,12 +109,13 @@ const enable = async (message: MessageAllowanceEnable, sender: Sender) => {
...
               PermissionMethodNostr.NOSTR_SIGNMESSAGE + "/" + kindId,
-              host
+              host,
+              response.data.blocked
             );
           });
         } else if (response.data.preset === NostrPermissionPreset.TRUST_FULLY) {
           Object.values(PermissionMethodNostr).forEach(async (permission) => {
-            await addPermissionFor(permission, host);
+            await addPermissionFor(permission, host, response.data.blocked);
```
[the remark this answers is not in the document]
Comment (pavanjoshi914) [reply src/extension/background-script/actions/nostr/enable.ts:118]: resolved

**Pull request merged**